packages/firestore/src/local/local_store_impl.ts

```diff
@@ -302,14 +316,36 @@ export function localStoreWriteLocally(
   const keys = mutations.reduce((keys, m) => keys.add(m.key), documentKeySet());
 
   let existingDocs: DocumentMap;
+  let mutationBatch: MutationBatch;
 
   return localStoreImpl.persistence
     .runTransaction('Locally write mutations', 'readwrite', txn => {
-      // Load and apply all existing mutations. This lets us compute the
-      // current base state for all non-idempotent transforms before applying
-      // any additional user-provided writes.
-      return localStoreImpl.localDocuments
-        .getDocuments(txn, keys)
+      // Figure out which keys do not have a remote version in the cache, this
+      // is needed to create the right overlay mutation: if no remote version
+      // presents, we do not need to create overlays as patch mutations.
+      // TODO(Overlay): Is there a better way to determine this? Document
+      //  version does not work because local mutations set them back to 0.
```

Comment:
```suggestion
      // TODO(Overlay): Is there a better way to determine this? Using the
      // document version does not work because local mutations set them back to 0.
```